src/lib.rs

```diff
@@ -47,115 +45,342 @@ pub enum Protection {
 pub struct Mmap {
     inner: MmapInner
 }
 
 impl Mmap {
-
-    /// Opens a file-backed memory map.
+    /// Change the `Protection` this mapping was created with.
     ///
-    /// The file must be opened with read permissions, and write permissions if
-    /// the supplied protection is `ReadWrite`. The file must not be empty.
-    pub fn open(file: &File, prot: Protection) -> Result<Mmap> {
-        let len = try!(file.metadata()).len();
-        if len > usize::MAX as u64 {
-            return Err(Error::new(ErrorKind::InvalidData,
-                                  "file length overflows usize"));
-        }
-        MmapInner::open(file, prot, 0, len as usize).map(|inner| Mmap { inner: inner })
+    /// This method only changes the protection of the underlying mapping,
+    /// but it doesn't make an `MmapMut` from an `Mmap`, use [`make_mut()`](#method.make_mut)
+    /// method for that.
+    ///
+    /// If you create a read-only file-backed mapping, you can **not** use this method to make the
+    /// mapping writeable. Remap the file instead.
+    ///
+    /// # Errors
+    ///
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
+    pub fn set_protection(&mut self, protection: Protection) -> Result<()> {
+        self.inner.set_protection(protection)
     }
 
-    /// Opens a file-backed memory map.
+    /// Change the `Protection` this mapping was created with to make it mutable.
     ///
-    /// The file must not be empty.
-    pub fn open_path<P>(path: P, prot: Protection) -> Result<Mmap>
-    where P: AsRef<Path> {
-        let file = try!(prot.as_open_options().open(path));
-        let len = try!(file.metadata()).len();
-        if len > usize::MAX as u64 {
-            return Err(Error::new(ErrorKind::InvalidData,
-                                  "file length overflows usize"));
+    /// If you create a read-only file-backed mapping, you can **not** use this method to make the
+    /// mapping writeable. Remap the file instead.
+    ///
+    /// # Errors
+    ///
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
+    ///
+    /// This method *also* returns `Err` with `ErrorKind` set to `InvalidInput` if the specified
+    /// protection does not allow the mapping to be mutable.
+    pub fn make_mut(mut self, protection: Protection) -> Result<MmapMut> {
+        self.inner.set_protection(protection)?;
+        match protection {
+            Protection::Read | Protection::ReadExecute => Err(Error::new(
+                ErrorKind::InvalidInput,
+                "Invalid protection for a mutable mapping",
+            )),
+            Protection::ReadWrite | Protection::ReadCopy => Ok(
```

Comment:
Yah, I think eventually we should, but we can punt for now and do it in a follow-up commit.